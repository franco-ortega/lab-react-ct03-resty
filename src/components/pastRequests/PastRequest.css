.PastRequest {
    background-color: gray;
    color: lightpink;
    margin: 10px;
    padding: 5px;
    border: solid 1px pink;
    border-radius: 10px;
    word-wrap: break-word;
}
