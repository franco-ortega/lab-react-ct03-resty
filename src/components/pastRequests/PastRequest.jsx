import React from 'react';
import PropTypes from 'prop-types';
import styles from './PastRequest.css';

export default function PastRequest({ method, url }) {
  return (
    <div className={styles.PastRequest}>
      <p>
        Method: {method}
      </p>
      <p>
        URL: {url}
      </p>
    </div>
  );
}

PastRequest.propTypes = {
  method: PropTypes.string.isRequired,
  url: PropTypes.string.isRequired
};




// export default class PastRequest extends Component {
//   render() {
//     return (
//       <div className={styles.PastRequest}>
//         <p>
//           Method:
//           {/* {this.state.method} */}
//         </p>
//         <p>
//           URL:
//           {/* {this.state.method} */}
//         </p>
//       </div>
//     );
//   }
// }
