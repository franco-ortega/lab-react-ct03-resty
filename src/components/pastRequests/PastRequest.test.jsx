import React from 'react';
import { render, cleanup } from '@testing-library/react';
import PastRequest from './PastRequest';

describe('PastRequest presentational component', () => {
  afterEach(() => cleanup());
  it('renders PastRequest', () => {
    const { asFragment } = render(<PastRequest
      method="GET"
      url="http://www.request.com"
    />);
    expect(asFragment()).toMatchSnapshot();
  });
});
