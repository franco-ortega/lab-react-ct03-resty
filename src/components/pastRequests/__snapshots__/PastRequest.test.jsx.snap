// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PastRequest presentational component renders PastRequest 1`] = `
<DocumentFragment>
  <div
    class="PastRequest"
  >
    <p>
      GET
    </p>
    <p>
      http://www.request.com
    </p>
  </div>
</DocumentFragment>
`;
